### ISSUE TITLE
regression: ICD for MongoDB and EnterpriseDB is reporting 'Unprocessable Entity'

### ISSUE THREAD

**Issue by shemau** [IBM-Cloud/terraform-provider-ibm#5263]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

Terraform 1.5.7 with IBM Provider 1.61.0

previously when creating MongoDB (6.0) and EnterpriseDB (12) with the ibm_database resource the following block was accepted:

```hcl
  group {
    group_id = "member"
    memory {
      allocation_mb = var.memory_mb
    }
    disk {
      allocation_mb = var.disk_mb
    }
    cpu {
      allocation_count = var.cpu_count
    }
    members {
      allocation_count = var.members
    }
  }
```

Note: The values specified varied by database type.

Now this block causes

```hcl
 ╷
│ Error: 1 error occurred:
│ 	* Unprocessable Entity
│ 
│ 
│ 
│   with module.enterprise_db.module.enterprise_db.ibm_database.enterprise_db,
│   on ../../main.tf line 52, in resource "ibm_database" "enterprise_db":
│   52: resource "ibm_database" "enterprise_db" {
│ 
╵
```

The error message is very poor. It gives no indication of what the problem is.
Trial and error removing blocks until the command did not error isolated the failure with the `group` block.
This used to work, but I have been unable to find a basic set of working values per

- https://cloud.ibm.com/docs/databases-for-mongodb?topic=databases-for-mongodb-pricing#mongodb-scale-member
- https://cloud.ibm.com/docs/databases-for-enterprisedb?topic=databases-for-enterprisedb-resources-scaling
- Using custom in the UI to determine default/min/max options

### Affected Resource(s)

* ibm_database

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# https://github.com/terraform-ibm-modules/terraform-ibm-icd-mongodb
```

### Debug Output

Nothing captured, the recreate is simple.

### Panic Output

There is no cras.

### Expected Behavior

Previous behaviour, the database instance should be provisioned

### Actual Behavior

An error occurs that the entity is unprocessable .

### Steps to Reproduce

Include the group block with minimal configurations (from the documentation link)

1. `terraform plan`

### Important Factoids

With the same terraform version and IBM Provider version the code still works for `databases-for-postgresql`. It is just failing with values that previously worked for mongo and enterprise.

### References

* #0000

**Comment by obai-1** [comment]
There should be more information after the message `Unprocessable Entity` which should help you narrow it down.

If not, could you set `TF_LOG=DEBUG`, re-run the command and send the log?

**Comment by shemau** [comment]
The entire error message is as above.

For mongo it is

```hcl
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: 1 error occurred:
│       * Unprocessable Entity
│ 
│ 
│ 
│   with module.mongodb.module.mongodb.ibm_database.mongodb,
│   on ../../main.tf line 49, in resource "ibm_database" "mongodb":
│   49: resource "ibm_database" "mongodb" {
│ 
```

Additional information:
The same error blocks destroy. Thus the infrastructure can not be torn down until the code is changed.

Trace logs to follow.

**Comment by shemau** [comment]
For `databases-for-enterprisedb` the error is

```hcl
{"errors":{"deployment.host_flavor":["enterprisedb is not available on multitenant"]}}: timestamp=2024-04-09T10:28:08.837+0100
2024-04-09T10:28:08.838+0100 [ERROR] provider.terraform-provider-ibm_v1.64.0: Response contains error diagnostic: tf_req_id=b260591d-07d4-53e9-dbae-0895053af2e0 @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR diagnostic_summary="1 error occurred:
        * Unprocessable Entity
```

In this case the `plan` is `standard`.

**Comment by omaraibrahim** [comment]
@shemau you need to set the `host_flavor` for `enterprisedb`. `host_flavors` can be found [here](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/database#host_flavor).

Please note that the `multitenant` option is not available for `enterprisedb`.

**Comment by omaraibrahim** [comment]
@shemau we've gone ahead and released a fix for this in `1.64.2`. Thank you for bringing this to our attention and we apologize for the inconvenience.

**Comment by shemau** [comment]
1.64.2 resolved this issue.
